src/lib/i18n/index.es

```diff
@@ -0,0 +1,211 @@
+   * If the ID is not specified returns a new instance.
+   * @param {string} id Instance ID. Optional.
+   */
+  constructor (id) {
+    let isValidId = typeof id === 'string'
+
+    if (!!isValidId && !!instances[id]) {
+      return instances[id]
+    } else if (!!isValidId && !instances[id]) {
+      instances[id] = this
+    }
+
+    /**
+     * Object where the translations are stored.
+     * @type {Object}
+     */
+    this._translations = {}
+  }
+
+  /**
+   * Sets the translations for a language.
+   * @param {string} lang Language identifier.
+   * @param {object} translations
+   * @returns {boolean} true if all went ok. false if not.
+   */
+  translations (lang, translations) {
+    if (!this._isValidLanguage(lang) || !this._areValidTexts(translations)) {
+      console.warn('i18n.translations: Invalid language or translations.')
+      return false
+    }
+
+    if (!this._languageExists(lang)) {
+      this._translations[lang] = {}
+    }
+
+    this._addTranslationsToLanguage(lang, translations)
+
+    return true
+  }
+
+  /**
+   * Sets the active language.
+   * @param {string} lang
+   * @returns {boolean}
+   */
+  setLanguage (lang) {
+    if (!this._isValidLanguage(lang)) {
+      console.warn('i18n.setLanguage: Invalid language.')
+      return false
+    }
+    language = lang
+    return true
+  }
+
+  /**
+   * returns the translated text if exist.
+   * @param {string} msgid Text identifier.
+   * @param {array} params
+   */
+  translate (msgid, params) {
+    let text = this._getTranslation(msgid)
+
+    if (!Array.isArray(params) || params.length === 0) {
+      return text
+    }
+
+    return this._replaceParams(text, params)
+  }
+
+  /**
+   * Checks if a language is valid.
+   * @param {string} lang
+   * @returns {boolean}
+   */
+  _isValidLanguage (lang) {
+    return typeof lang === 'string'
+  }
+
+  /**
+   * Checks if a language exists.
+   * @param {string} lang
+   * @returns {boolean}
+   */
+  _languageExists (lang) {
+    return this._isValidLanguage(lang) && this._areValidTexts(this._translations[lang])
+  }
+
+  /**
+   * Checks if the translations are valid.
+   * @param {object} translations
+   * @returns {boolean}
+   */
+  _areValidTexts (translations) {
+    return typeof translations === 'object' && translations !== null
+  }
+
+  /**
+   * Adds new translation to the language.
+   * @param {string} lang
+   * @param {object} translations
+   * @returns {boolean}
+   */
+  _addTranslationsToLanguage (lang, translations) {
+    for (let key in translations) {
+      if (translations.hasOwnProperty(key)) {
+        this._translations[lang][key] = translations[key]
+      }
+    }
+  }
+
+  /**
+   * Returns the translation for the given msgid.
+   * @param {string} msgid
+   * @returns {string}
+   */
+  _getTranslation (msgid) {
+    if (this._languageExists(language) && typeof this._translations[language][msgid] !== 'undefined') {
```

Comment:
Is the same the following?
```this._translations[language][msgid] !== 'undefined'```
and
```!!this._translations[language][msgid]```
considering that either it's ever been defined or it's full, but it's never being null?